app/models.py

```diff
@@ -46,10 +46,11 @@ class Hospital(db.Model):
     zipcode = db.Column(db.String(128), default="")
     name = db.Column(db.String(140), default="")
     credit = db.Column(db.Integer, default=0)
-    wants = db.relationship('Wants', backref='author1', lazy='dynamic')
-    has = db.relationship('Has', backref='author2', lazy='dynamic')
-    users = db.relationship('User', backref='author3', lazy='dynamic')
-    exchange = db.relationship('Exchange', backref='author7', lazy='dynamic')
+    wants = db.relationship('Wants', backref='hospital')
+    has = db.relationship('Has', backref='hospital')
+    users = db.relationship('User', backref='hospital')
+    exchange1s = db.relationship('Exchange', backref='hospital1_ref', foreign_keys='Exchange.hospital1')
```

Comment:
We have to have two relationships here (I think--maybe you could be clever and get away with one) because there are two foreign keys on the `Exchange` table.

Reply:
These (along with a few other backrefs) are called `thing_ref` because `thing` already existed on the target table (only `Exchange` IIRC).